fw/http_parser.c

```diff
@@ -9558,6 +9568,13 @@ tfw_h2_parse_req_hdr(unsigned char *data, unsigned long len, TfwHttpReq *req,
 				__FSM_H2_FIN(Req_HdrUser_AgentV, 10,
 					     TFW_TAG_HDR_USER_AGENT);
 			__FSM_H2_OTHER_n(4);
+		/* upgrade */
+		case TFW_CHAR4_INT('u', 'p', 'g', 'r'):
+			if (unlikely(!__data_available(p, 7)))
+				__FSM_H2_NEXT_n(Req_HdrUpgr, 4);
+			if (C4_INT(p + 3, 'r', 'a', 'd', 'e'))
+				__FSM_H2_DROP(Req_HdrUpgrade);
+			__FSM_H2_OTHER_n(4);
```

Comment:
Could you please add a comment with reference to RFC 9113 8.2.2 that we tread an HTTP/2 message having `Upgrade` header as malformed? The same for the `Proxy-Connection` above